TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+package org.firstinspires.ftc.teamcode.hardware;
+
+import static org.firstinspires.ftc.teamcode.utility.Config.ENCODER_PER_INCH;
+
+import androidx.annotation.Nullable;
+
+import com.qualcomm.robotcore.hardware.DcMotor;
+import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.IMU;
+
+import org.firstinspires.ftc.robotcore.external.Supplier;
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
+import org.firstinspires.ftc.teamcode.utility.Api;
+
+/**
+ * Class that contains the entire drivebase for the robot. Contains methods to move the robot both in auto and in teleop.
+ */
+public class Drivebase {
+    private final DcMotor frdrive;
+
+    private final DcMotor fldrive;
+    private final DcMotor brdrive;
+    private final DcMotor bldrive;
+    private final IMU imu;
+    private final Supplier<Boolean> opModeIsActive;
+
+
+    /**
+     * Uses `hardwareMap` to initialize the motors and imu.
+     *
+     * @param hardwareMap    The hardware map to use for initialization.
+     * @param opModeIsActive Supplier that returns false if the program needs to stop.
+     */
+    public @Api Drivebase(HardwareMap hardwareMap, Supplier<Boolean> opModeIsActive) {
```

Comment:
What is the Supplier<Boolean> doing here?

Reply:
We need it so that if the driver presses stop, our auto exits gracefully instead of looping forever.